java-squid/src/main/java/org/sonar/java/resolve/BytecodeVisitor.java

```diff
@@ -318,11 +321,15 @@ public ReadGenericSignature() {
 
     @Override
     public void visitFormalTypeParameter(String name) {
-      typeVariableSymbol = new JavaSymbol.TypeVariableJavaSymbol(name, classSymbol);
-      classSymbol.typeParameters.enter(typeVariableSymbol);
-      classSymbol.addTypeParameter((JavaType.TypeVariableJavaType) typeVariableSymbol.type);
-      bounds = Lists.newArrayList();
-      ((JavaType.TypeVariableJavaType) typeVariableSymbol.type).bounds = bounds;
+      List<JavaSymbol> lookup = classSymbol.typeParameters.lookup(name);
+      Preconditions.checkState(lookup.size() == 1, "found "+lookup.size());
+      typeVariableSymbol = (JavaSymbol.TypeVariableJavaSymbol) lookup.iterator().next();
+      bounds = ((JavaType.TypeVariableJavaType) typeVariableSymbol.type).bounds;
+//      typeVariableSymbol = new JavaSymbol.TypeVariableJavaSymbol(name, classSymbol);
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)This block of commented-out lines of code should be removed.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)